src/stores/token.ts

```diff
@@ -117,4 +123,23 @@ export default class Token extends ContractStore {
       .call()
       .then(this.updateBalance);
   }
+
+  /*
+  * Checks transfer allowance for a given spender. If allowance is not enough to transfer a given value,
+  * initiates an approval transaction for 2^256 units
+  * @param spender Account to approve transfer for
+  * @param value Minimal amount of allowance a spender should have
+  * @returns Promise resolved when allowance is enough for the transfer
+  */
+  private maybeApprove(spender: string, value: BigNumber) {
+    return this.iContract.methods.allowance(this.account.address, spender).call().then(allowance => {
+      if (Number(allowance) >= value.toNumber()) return;
+
+      const tx = this.iContract.methods
+        .approve(spender, new BigNumber(2 ** 255))
```

Comment:
Why not `value − allowance`?

Reply:
so that we have to execute `approve` only one time. Next deposit/bet will require just on tx

Reply:
Yeah, but it looks like too invasive, no? It's ok for rinkeby, but not for mainnet

Reply:
I thought so too, but some projects already doing that. For example, I was trading on https://radarrelay.com/ and they set the same DAI and WETH allowances for me. Not good (you have to keep trust), but better UX

Reply:
Yeah, let's keep it for now.

I have idea for UI/UX, but i need to sleep with it :-)

Reply:
> I have idea for UI/UX, but i need to sleep with it :-)

me too :)